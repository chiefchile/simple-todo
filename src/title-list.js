import { default as React } from 'react';

export const TitleList = (props) => {

	return (
		<ul>
			{props.titles.map(title => 
			<li key={title.noteId}>
				<a href="#" onClick={() => props.onClick(title.noteId)}>{title.title}</a>
			</li>)}
		</ul>
	)
	
}