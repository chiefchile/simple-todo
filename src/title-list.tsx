import { default as React } from 'react';
import Title from './title';
import Result from './result';

interface Props {
	titles: Title[],
	onClick(_id: string | undefined, result: Result | null): void
}

export const TitleList = (props: Props) => {

	return (
		<div>
			<h4>Notes</h4>
			<ul>
				{props.titles.map((title: Title) =>
					<li key={title._id}>
						<a href="#" onClick={() => props.onClick(title._id, null)}>{title.title}</a>
					</li>)}
			</ul>
		</div>
	)

}