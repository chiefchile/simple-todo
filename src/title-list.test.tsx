import React from 'react';
import { TitleList } from './title-list';
import renderer from 'react-test-renderer';
import Title from './title';



it('renders correctly', () => {
  const titles: Title[] = [{title: 'Title2', _id: '1'}];
  const tree = renderer
    .create(<TitleList titles={titles} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});