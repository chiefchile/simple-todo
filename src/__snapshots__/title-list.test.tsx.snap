// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h4>
    Notes
  </h4>
  <ul>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Title2
      </a>
    </li>
  </ul>
</div>
`;
